tenseal/binding.cpp

```diff
@@ -134,10 +134,14 @@ PYBIND11_MODULE(_tenseal_cpp, m) {
         .def("dot_", &CKKSVector::dot_product_plain_inplace)
         .def("sum", &CKKSVector::sum)
         .def("sum_", &CKKSVector::sum_inplace)
-        .def("matmul", &CKKSVector::matmul_plain, py::arg("matrix"))
-        .def("matmul_", &CKKSVector::matmul_plain_inplace, py::arg("matrix"))
-        .def("mm", &CKKSVector::matmul_plain, py::arg("matrix"))
-        .def("mm_", &CKKSVector::matmul_plain_inplace, py::arg("matrix"))
+        .def("matmul", &CKKSVector::matmul_plain, py::arg("matrix"),
+             py::arg("n_batches") = 0)
+        .def("matmul_", &CKKSVector::matmul_plain_inplace, py::arg("matrix"),
+             py::arg("n_batches") = 0)
+        .def("mm", &CKKSVector::matmul_plain, py::arg("matrix"),
+             py::arg("n_batches") = 0)
+        .def("mm_", &CKKSVector::matmul_plain_inplace, py::arg("matrix"),
+             py::arg("n_batches") = 0)
```

Comment:
I know this sounds philosophic lol but I think the user doesn't need to know that we process this in batches or not, the batch thing is more specific for our matmul implementation, it can be the parameter name for diagonal_vector_matmul function for example, but the user should only get that the operation is done via `n` jobs, that's why I would prefer the job terminology exposed to the user in the API